app/code/Magento/Email/Model/Template/Filter.php

```diff
@@ -237,7 +229,6 @@ public function __construct(
         $this->_layoutFactory = $layoutFactory;
         $this->_appState = $appState;
         $this->urlModel = $urlModel;
-        $this->emogrifier = $emogrifier;
         $this->cssInliner = $cssInliner ?: \Magento\Framework\App\ObjectManager::getInstance()
```

Comment:
Considering that the contractor parameters have to be changed anyway, I think we should use this opportunity to properly add the parameters that are currently added in a backward compatible way.

Can you please change such optional parameters to required for this class and all classes that are extending the constructor.